src/screens/NationDetailsContainer/NationDetailsScreen/index.js

```diff
@@ -217,31 +157,121 @@ class NationDetailsScreen extends Component {
+            iconSource={AssetsImage.Actions.delete}
+            title={i18n.t('screens.nations.toolbar.delete')}
+            disable={false}
+            onPress={this.props.deleteDraft}
+          />
+          <NationActionButton
+            iconSource={AssetsImage.Actions.submit}
+            title={i18n.t('screens.nations.toolbar.submit')}
+            disable={!nationIsValid(nation)}
+            onPress={this.props.submitDraft}
+          />
+        </View>
       );
     }
+
+    return (
+      <View style={styles.fakeBottomBar}>
+        <NationActionButton
+          iconSource={AssetsImage.Actions.chat}
+          title={i18n.t('screens.nations.toolbar.chat')}
+          disable={false}
+          onPress={this.props.openNationChat}
+        />
+        <NationActionButton
+          iconSource={AssetsImage.Actions.map}
+          title={i18n.t('screens.nations.toolbar.map')}
+          disable
+          onPress={() => {}}
+        />
+        <NationActionButton
+          iconSource={AssetsImage.Actions.join}
+          title={i18n.t('screens.nations.toolbar.join')}
+          disable={NationDetailsScreen.disableJoinButton(nation)}
+          onPress={this.props.joinNation}
+        />
+        <NationActionButton
+          iconSource={AssetsImage.Actions.leave}
+          title={i18n.t('screens.nations.toolbar.leave')}
+          disable={NationDetailsScreen.disableLeaveButton(nation)}
+          onPress={this.props.leaveNation}
+        />
+      </View>
+    );
+  }
+  render() {
+    const nation = openedNation(this.props);
+
+    if (!nation) {
+      this.props.navigator.pop();
+      return <BackgroundImage />;
+    }
+
+    const status = resolveStatus(nation);
+    const statusDescription = (status !== null ? i18n.ifExists(`screens.nationDetails.statusDescription.${status.key}`) : '');
+
+    return (
+      <View style={styles.screenContainer}>
+        <BackgroundImage />
+        <FakeNavigationBar />
+        <View style={styles.bodyContainer}>
+          <View style={styles.titleContainer}>
+            <View style={styles.titleBarLarge}>
+              <Text style={styles.largeTitle}>{nation.nationName}</Text>
+            </View>
+          </View>
+
+          <ScrollView>
+            {statusDescription !== '' && NationDetailsScreen.buildStatusPanel(statusDescription)}
+
+            {NationDetailsScreen.buildAboutView(nation)}
+            {NationDetailsScreen.buildCitizenPanel(nation)}
+            {NationDetailsScreen.buildGovernmentalStructureView(nation)}
+            {NationDetailsScreen.buildFactsView(nation)}
+          </ScrollView>
+        </View>
+        {this.buildTabBar()}
+      </View>
+    );
   }
 }
 
-NationDetailsScreen.propTypes = {
-  isDraft: PropTypes.bool,
-  joinNation: PropTypes.func,
-  leaveNation: PropTypes.func,
-  deleteDraft: PropTypes.func,
-  submitDraft: PropTypes.func,
-  openNationChat: PropTypes.func,
+NationDetailsScreen.defaultProps = {
+  isDraft: null,
```

Comment:
Please don't pass null to props that has required type (boolean in that case). If there is no correct default value, you can just remove it from there.

Reply:
Done